src/app/modules/overlay-slide-ins/overlay-slide-in.service.ts

```diff
@@ -0,0 +1,130 @@
+import {
+  Overlay,
+  OverlayConfig,
+} from '@angular/cdk/overlay';
+import {
+  ComponentPortal,
+} from '@angular/cdk/portal';
+import {
+  computed,
+  Injectable,
+  Injector,
+  signal,
+  Type,
+} from '@angular/core';
+import { UntilDestroy, untilDestroyed } from '@ngneat/until-destroy';
+import { UUID } from 'angular2-uuid';
+import { Subject } from 'rxjs';
+import { OverlayContainerComponent } from 'app/modules/overlay-slide-ins/components/overlay-container/overlay-container.component';
+import { SlideInOverlayRef } from 'app/modules/overlay-slide-ins/slide-in-overlay-ref';
+import { SLIDE_IN_DATA } from 'app/modules/overlay-slide-ins/slide-in.tokens';
+
+interface OverlayInstance<T> {
+  overlayId: string;
+  slideInRef: SlideInOverlayRef;
+  component: Type<T>;
+  config: { data?: unknown; wide?: boolean };
+  close$: Subject<T | undefined>;
+}
+
+@UntilDestroy()
+@Injectable({ providedIn: 'root' })
+export class OverlaySlideInService {
+  private overlays = signal<OverlayInstance<unknown>[]>([]);
+  openOverlays = computed(() => this.overlays()?.length);
+
+  constructor(
+    private overlay: Overlay,
+    private injector: Injector,
+  ) {}
+
+  open<C, T = unknown>(
```

Comment:
Previous implementation ensured that dataand return value was strongly typed. See ComponentInSlideIn